Sources/OpenAPIRuntime/URICoder/Parsing/URIParser.swift

```diff
@@ -326,4 +372,35 @@ extension String.SubSequence {
         }
         return finalize()
     }
+    
+    
+    /// Accumulates characters from the `startingCharacter` character provided,
+    /// until the `endingCharacter` is reached. Moves the underlying startIndex.
+    /// - Parameters:
+    ///   - startingCharacter: A character to start with.
+    ///   - endingCharacter: A character to stop at.
+    /// - Returns: The accumulated substring.
+    fileprivate mutating func parseBetweenCharacters(startingCharacter: Character, endingCharacter: Character) -> Self {
```

Comment:
This function seems to overlap significantly with `parseUpToCharacterOrEnd(_:)`, the only difference is that we move the index forward to the first occurrence of `startingCharacter` first. Do we need the duplication, or can we implement one in terms of the other?

Reply:
You mean keeping just one of the functions and add an extra param in order to determine its functionality?

Reply:
What'd be the extra param needed? I also agree we should be able to use the existing function.